openedx/core/djangoapps/catalog/utils.py

```diff
@@ -55,8 +56,14 @@ def get_programs(uuid=None):
             logger.warning(missing_details_msg_tpl.format(uuid=uuid))
 
         return program
+    uuids = []
     if waffle.switch_is_active('get-multitenant-programs'):
-        uuids = cache.get(SITE_PROGRAM_UUIDS_CACHE_KEY_TPL.format(domain=get_current_site().domain), [])
+        # TODO: LEARNER-1146 we have to cover this in test cases, when we remove the waffle check.
+        if get_current_site():
+            uuids = cache.get(SITE_PROGRAM_UUIDS_CACHE_KEY_TPL.format(domain=get_current_site().domain), [])
+        else:
+            for domain in Site.objects.all().values_list('domain', flat=True):
+                uuids.extend(cache.get(SITE_PROGRAM_UUIDS_CACHE_KEY_TPL.format(domain=domain), []))
```

Comment:
From a clean code perspective, it would be better if we passed the site into `get_programs`, but I understand that would be a much larger change. Do you have time to take that on? 

If not, I would suggest adding a TODO explaining that this is a temporary fix which will be revisited with LEARNER-1146.